moveit_plugins/moveit_ros_control_interface/src/controller_manager_plugin.cpp

```diff
@@ -419,6 +452,38 @@ class MoveItControllerManager : public moveit_controller_manager::MoveItControll
     }
     return true;
   }
+  /**
+   * \brief fixChainedControllers modifies ListControllers service response if it contains chained controllers.
+   * Since chained controllers cannot be written to directly, they are removed from the response and their interfaces
+   * are propagated back to the first controller with a non-chained input
+   */
+  bool fixChainedControllers(std::shared_ptr<controller_manager_msgs::srv::ListControllers::Response>& result)
+  {
+    std::unordered_map<std::string, controller_manager_msgs::msg::ControllerState*> controller_name_map;
```

Comment:
Prob should make this a unique_ptr for safety's sake.
```suggestion
    std::unordered_map<std::string, std::unique_ptr<controller_manager_msgs::msg::ControllerState>> controller_name_map;
```